### PR TITLE
NPC refactor

### PR DESCRIPTION
**Pull request by metalgearsloth**
- Prototypes everywhere
- More ECS
- Better debugging tools (TBC cough)

TODO:
- [x] Docs on the dev wiki

Alright so utility AI replaced with HTN. It's essentially just a tree of primitive tasks (nodes) and compound tasks (leaves). From a root compound task it gets expanded upon trying to find the first valid "plan". Compound tasks get expanded further and further until we only have primitive tasks remaining (a "plan"). A compound task can have N branches and each branch can have N sequential tasks. Each primitive task has an "operator" (the actual code that gets run) attached to it.
Easiest way to see it in game is run `showhtn` and it shows the BTR + tasks being run. Plan A is considered better than plan B if it expands earlier (branch traversal record / method traversal record).

I've tried to move as much as I can to prototypes which also means it should be more re-usable (as exemplified by the linecount going down a lot... plus the removal of the old blackboard).

I moved the blackboard (AKA how the NPC thinks about data when planning) to just 1 class with an internal dictionary. It's kinda scuffed but I also couldn't think of anything easier as I need to be able to represent data changing during planning so willing to change it to whatever.

This also includes rudimentary dedicated melee + ranged + steering systems that will be built upon later. The reason they're in their own systems and not operators is because they'll require more detailed co-operation between NPCs (e.g. to avoid stacking) and it's just easier to do it in a dedicated system.

Pathfinding is largely untouched for now but it will be the first thing I come back to after my next PR after I suss out recast. I have collision avoidance somewhat working from an external library but want to split it out and polish it more before using it.

Docs are at https://docs.spacestation14.io/e/en/NPCs

I scope crept upwards and then back down for now so I may have forgor to delete some things.

:cl:
- tweak: Pets will no longer try to run away when buckled or pulled.
- tweak: Space carps will now try to follow their dragon.
- add: Turrets now exist.

### CODE REVIEW

**Comment by metalgearsloth** [comment]
Just needs the collision avoidance split out and dev docs and then it's good for review (the line count is mostly just the debug map).

**Review by moonheart08 — APPROVED** [review]

File: Resources/Textures/Objects/Weapons/Guns/turrets.rsi/meta.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "version": 1,
+  "size": {
+    "x": 32,
+    "y": 32
+  },
```
Comment (moonheart08) [inline Resources/Textures/Objects/Weapons/Guns/turrets.rsi/meta.json:6]: Missing copyright.

**Review by moonheart08 — CHANGES_REQUESTED** [review]

**Review by moonheart08 — APPROVED** [review]

File: Content.Server/NPC/Pathfinding/IPathfindingGraph.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+namespace Content.Server.NPC.Pathfinding;
+
+public interface IPathfindingGraph
+{
+
+}
```
Comment (mirrorcult) [inline Content.Server/NPC/Pathfinding/IPathfindingGraph.cs:6]: empty?

**Review by mirrorcult — DISMISSED** [review]
a-ok to merge once engine stuff is figured out

**Comment by Partmedia** [comment]
Clicking the link to the docs in the PR description gives me: "Unauthorized You cannot create the page." Has this moved elsewhere?

**Pull request merged**